app/Helpers/database/leaderboard.php

```diff
@@ -49,6 +49,7 @@ function SubmitLeaderboardEntry(
         timestamp: $timestamp,
     );
 
+    // First check if there's an existing entry (including soft-deleted)
```

Comment:
The soft-deleted record should be found by the `withTrashed` clause. But there's still a possibility that two requests will both get to the `else` before either finishes adding the row.